### PR TITLE
Remove SequenceActivities.

### LINKED ISSUE #15632: Properly cancel/finish Child-/SequenceActivities instead of nulling them
We have several bugs caused by unconditionally nulling or replacing activities, instead of properly canceling/finishing the current one first:

#9404
#10223
#10224
#15631
and probably quite a few more.

To fix this, we'll have to update the activity code to support propagating cancellation through activity sequences (including ChildActivity sequences).
For the sake of consistency and any potential cases that might need it, we should probably do the same for `OnActorDispose` introduced by #15622.

### PR DESCRIPTION
**Pull request by tovl**
As discussed on IRC.

Fixes #15632

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Resupply.cs [hunk 1]
```diff
@@ -99,23 +99,19 @@ public override bool Tick(Actor self)
...
-
-				var moveActivities = ActivityUtils.SequenceActivities(movement.ToArray());
+					QueueChild(move.MoveTo(targetCell, host.Actor));
 
 				var delta = (self.CenterPosition - host.CenterPosition).LengthSquared;
 				var transport = transportCallers.FirstOrDefault(t => t.MinimumDistance.LengthSquared < delta);
 				if (transport != null)
 					transport.RequestTransport(self, targetCell);
 
-				QueueChild(moveActivities);
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Activities/Resupply.cs]: We need to test this carefully. I'm not sure anymore, but it's possible this had to be queued after the transport request for a reason.

Comment (pchote) [reply OpenRA.Mods.Common/Activities/Resupply.cs]: #16778 fixed the carryall logic messing with the activity queue, so this should be safe.

**Review by pchote — APPROVED** [review]

File: OpenRA.Mods.Common/Traits/Mobile.cs [hunk 2]
```diff
@@ -319,10 +319,9 @@ public void Nudge(Actor self, Actor nudger, bool force)
 				if (cellInfo != null)
 				{
 					self.CancelActivity();
-					var notifyBlocking = new CallFunc(() => self.NotifyBlocker(cellInfo.Cell));
-					var waitFor = new WaitFor(() => CanEnterCell(cellInfo.Cell));
-					var move = new Move(self, cellInfo.Cell);
-					self.QueueActivity(ActivityUtils.SequenceActivities(notifyBlocking, waitFor, move));
+					self.QueueActivity(new CallFunc(() => self.NotifyBlocker(cellInfo.Cell)));
+					self.QueueActivity(new WaitFor(() => CanEnterCell(cellInfo.Cell)));
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: Out of scope for this PR, but for future reference: the two above look like they should become child activities of `Move` at some point, or even not be their own activities at all: we could do `self.NotifyBlocker` in `OnFirstRun` and make `Tick` `return false` early if `!CanEnterCell`.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/e6c878e0bf50904ae010b9d4abc8c19adcd4ad97...4560a9999b3dbede9c8b3e57b6aa6c401d251312

**Pull request merged**